src/modules/kubernetes.rs

```diff
@@ -80,30 +86,44 @@ fn get_kube_ctx_component(filename: path::PathBuf, current_ctx: &str) -> Option<
     Some(ctx_components)
 }
 
-fn get_kube_user<'a>(config: &'a KubernetesConfig, kube_user: &'a str) -> Cow<'a, str> {
-    return get_alias(&config.user_aliases, kube_user).unwrap_or(Cow::Borrowed(kube_user));
-}
-
-fn get_kube_context_name<'a>(config: &'a KubernetesConfig, kube_ctx: &'a str) -> Cow<'a, str> {
-    return get_alias(&config.context_aliases, kube_ctx).unwrap_or(Cow::Borrowed(kube_ctx));
-}
-
-fn get_alias<'a>(
-    aliases: &'a HashMap<String, &'a str>,
-    alias_candidate: &'a str,
-) -> Option<Cow<'a, str>> {
-    if let Some(val) = aliases.get(alias_candidate) {
-        return Some(Cow::Borrowed(val));
+fn get_aliased_name<'a>(
+    pattern: Option<&'a str>,
+    current_value: Option<&String>,
+    alias: Option<&'a str>,
+) -> Option<String> {
+    let replacement = match (&alias, &current_value) {
+        (None, None) => return None,
+        (Some(alias_value), _) => (*alias_value).to_string(),
+        (None, Some(value)) => (*value).clone(),
+    };
+    let Some(pattern) = pattern else {
+        // If user pattern not set, treat it as a match-all pattern
+        return Some(replacement);
+    };
+    let Some(value) = current_value else {
+        // If a pattern is set, but we have no value, there is no match
+        return None;
+    };
+    if value.as_str() == pattern {
+        return Some(replacement);
     }
-
-    return aliases.iter().find_map(|(k, v)| {
-        let re = regex::Regex::new(&format!("^{k}$")).ok()?;
-        let replaced = re.replace(alias_candidate, *v);
-        match replaced {
-            Cow::Owned(replaced) => Some(Cow::Owned(replaced)),
-            _ => None,
+    let re = match regex::Regex::new(&format!("^{pattern}$")) {
+        Ok(re) => re,
+        Err(error) => {
+            log::warn!(
+                "Could not compile regular expression `{}`:\n{}",
+                &format!("^{pattern}$"),
+                error
```

Comment:
This woul have been a nice let-else example, but I haven'T found a way to access the error (to print) in that form :-(